File: C++/Task480.cpp
#include <iostream>
using namespace std;
double min(double a, double b, double c)
{
    double result;
    if (a < b)
    {
        result = a;
    }
    else
    {
        result = b;
    }
    if (c < result)
    {
        result = c;
    }
    return result;
}
double max(double a, double b)
{
    return a > b ? a : b;
}
// основна програма
int main()
{
    double s, t, x, y, p, p1, p2;
    cout << "Enter input values for s,t,x,y\n";
    cin >> s >> t >> x >> y;
    // Значення виразу знайдемо як суму двох частин:
    //   віднімемо перші два дороби і наступні два
    p1 = min(t, x, y) / min(s, t, y) -
         max(t, s) / min(s, t, x);
    p2 = max(x, y) / min(s, x, y) -
         max(t + s, x + y) / max(x + s, t + y);
    p = p1 + p2;
    printf("p =  %12.5f\n", p);
}

File: C#/Task480.cs
using System;

class Program
{
    static double Min(double a, double b, double c)
    {
        double result = a < b ? a : b;
        if (c < result)
            result = c;
        return result;
    }

    static double Max(double a, double b)
    {
        return a > b ? a : b;
    }

    static void Main()
    {
        double s, t, x, y;

        Console.WriteLine("Enter input values for s, t, x, y");
        string[] parts = Console.ReadLine().Split();
        s = double.Parse(parts[0]);
        t = double.Parse(parts[1]);
        x = double.Parse(parts[2]);
        y = double.Parse(parts[3]);

        double p1 = Min(t, x, y) / Min(s, t, y) -
                    Max(t, s) / Min(s, t, x);

        double p2 = Max(x, y) / Min(s, x, y) -
                    Max(t + s, x + y) / Max(x + s, t + y);

        double p = p1 + p2;

        Console.WriteLine($"p = {p,12:F5}");
    }
}
